llvm/include/llvm/ADT/STLExtras.h

```diff
@@ -1704,6 +1704,11 @@ auto sum_of(R &&Range, E Init = E{0}) {
   return accumulate(std::forward<R>(Range), std::move(Init));
 }
 
+/// Return a range where each value of `R` is static_cast to `T`
+template <typename T, typename R> auto static_cast_to(R &&Range) {
```

Comment:
I think the idea makes sense but I wonder if we could pick a more descriptive function name, similar to other range functions

Reply:
One alternative would be not to add a range function and instead lift static cast to a function object like we did for `llvm::IsaPred<T>`. This way we can support variants based on `dyn_cast`/`dyn_cast_if_present`/`cast`.

I think that's a more general solution that composes with other functions like `map_to_vector`

Reply:
`map_range(Foo, StaticCastTo<unsigned>)`
`map_range(Bar, DynCastTo<IntegerAttr>)`